common/keyboard.go

```diff
@@ -310,6 +311,7 @@ func split(keys string) []string {
 	var ss string
 	justAdded := true
 	for _, s := range keys {
+		//nolint:gocritic
 		if string(s) == "+" && ss != "" {
 			kk = append(kk, ss)
 			ss = ""
```

Comment:
The linter is right here. It's better to convert the if-else chain into a switch.

```go
for _, k := range keys {
	switch {
	case k == '+' && ss != "":
		...
...
```

And, you can use [`strings.Builder`](https://stackoverflow.com/questions/1760757/how-to-efficiently-concatenate-strings-in-go/47798475#47798475) to append runes without converting them.

Reply:
Nice, I hadn't realised that it was idiomatic to write them as switch statements. Source: https://go.dev/doc/effective_go#switch

Reply:
Isn't it more like this?

```go
func split(keys string) []string {
	var (
		kk = make([]string, 0, len(keys))
		s  strings.Builder
	)
	for _, r := range keys {
		if r == '+' && s.Len() > 0 {
			kk = append(kk, s.String())
			s.Reset()
		} else {
			s.WriteRune(r)
		}
	}
	kk = append(kk, s.String())

	return kk
}
```

See: https://go.dev/play/p/fJoqZIMTBF5